<?xml version="1.0" ?>
<launch>
  <arg name="sim" default="true"/>
  <arg name="real_robot" value="false" />

  <arg name="gui" default="false" />
  <arg name="gazebo_gui" default="false" />

  <arg name="server_port" default ="50051"/>
  <arg name="action_cycle_rate" default="25" doc="Rate at which new commands are published to the robot controller"/>

  <!-- Launch uwrt_arm_bringup -->
  <include file="$(find uwrt_arm_bringup)/launch/arm_bringup.launch">
    <arg name="sim"   value="$(arg sim)"/>
  </include>

  <!-- Launch uwrt_arm_bringup -->
  <include file="$(find uwrt_arm_rover_simulation)/launch/empty_world.launch"/>

  <!-- Launch RViz -->
<!--   <node if="$(arg gui)" name="$(anon rviz)" pkg="rviz" type="rviz" respawn="false" -->
<!--   args="-d $(find uwrt_robot_server)/rviz/uwrt_sim_robot_server.rviz" output="screen"> -->
<!--   </node> -->
  
  <!-- Launch Command Handler  -->
  <node name="joint_position_command_handler" pkg="uwrt_robot_server" type="joint_position_command_handler.py" respawn="false" output="screen" >
    <param name="real_robot"        value="$(arg real_robot)"/>
    <param name="action_cycle_rate" value="$(arg action_cycle_rate)"></param>
  </node>

  <!--Launch gRPC server -->
  <node name="robot_server" pkg="uwrt_robot_server" type="robot_server.py" respawn="false" output="screen" >
    <param name="server_port"                 value="$(arg server_port)"></param>
    <param name="real_robot"                  value="$(arg real_robot)"/>
    <param name="action_cycle_rate"           value="$(arg action_cycle_rate)"></param>
  </node>

</launch>
